File: _includes/code/hello.go
package main

import "fmt"

// Main function for the action
func Main(obj map[string]interface{}) map[string]interface{} {
	name, ok := obj["name"].(string)
	if !ok {
		name = "stranger"
	}
	fmt.Printf("name=%s\n", name)
	msg := make(map[string]interface{})
	msg["msg"] = "Hello, " + name + "!"
	return msg
}

File: _includes/code/Hello.java
import com.google.gson.JsonObject;

public class Hello {
    public static JsonObject main(JsonObject args){
        String name;

        try {
            name = args.getAsJsonPrimitive("name").getAsString();
        } catch(Exception e) {
            name = "stranger";
        }

        JsonObject response = new JsonObject();
        response.addProperty("greeting", "Hello " + name + "!");
        return response;
    }
}
